File: 0985. Sum of Even Numbers After Queries/Solution.py
from typing import List


class Solution:
    def sumEvenAfterQueries(self, A: List[int], queries: List[List[int]]) -> List[int]:
        accum = sum(x for x in A if x % 2 == 0)
        ret = []
        for val, idx in queries:
            if A[idx] % 2 == 0:
                accum -= A[idx]
            A[idx] += val
            if A[idx] % 2 == 0:
                accum += A[idx]
            ret.append(accum)
        return ret

File: 0985. Sum of Even Numbers After Queries/Solution.cpp
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> sumEvenAfterQueries(vector<int> &A, vector<vector<int>> &queries)
    {
        int sum = 0;
        for (const int &x : A)
            if (!(x & 1))
                sum += x;
        
        vector<int> ret;
        for (const vector<int> &query : queries) {
            int &val = A[query[1]];
            if (!(val & 1))
                sum -= val;
            val += query[0];
            if (!(val & 1))
                sum += val;
            ret.emplace_back(sum);
        }
        return ret;
    }
};
